apps/src/templates/projects/PersonalProjectsNameCell.jsx

```diff
@@ -0,0 +1,52 @@
+import React, {Component, PropTypes} from 'react';
+import {connect} from 'react-redux';
+import {tableLayoutStyles} from "../tables/tableConstants";
+import {startRenamingProject} from './projectsRedux';
+
+const styles = {
+  inputBox: {
+    width: 225,
+  },
+};
+
+class PersonalProjectsNameCell extends Component {
+  static propTypes = {
+    projectId: PropTypes.string.isRequired,
+    projectType: PropTypes.string.isRequired,
+    projectName: PropTypes.string.isRequired,
+    isEditing: PropTypes.bool,
+    updatedName: PropTypes.string,
+    startRenamingProject: PropTypes.func.isRequired,
+  };
+   onChangeName = (e) => {
+    this.props.startRenamingProject(this.props.projectId, e.target.value);
```

Comment:
...however in this case it looks like you are overloading `startRenamingProject` to also update the new name of the project after the user has modified it, which is a bit misleading.

I would recommend simplifying the `startRenamingProject` to only make the field editable (without changing the name), and use a separate prop / redux action here to update the name as the user types. this one could be called 'updateProjectName'.